velox/exec/tests/NestedLoopJoinTest.cpp

```diff
@@ -231,29 +312,68 @@ TEST_F(NestedLoopJoinTest, zeroColumnBuild) {
   assertQuery(op, "SELECT * FROM t");
 }
 
-// Test multi-threaded build and probe sides.
-TEST_F(NestedLoopJoinTest, parallelism) {
-  // Setup 5 threads for build and probe. Each build thread gets 3 identical
-  // rows of input from the Values operator. The build thread that finishes last
-  // combines data from all other threads making it 3x5=15 rows and puts them
-  // into the NestedLoopJoinBridge. All probe threads get 2 identical ros of
-  // input from the Values operator and join them with 15 rows of build side
-  // data from the bridge. Each probe thread is expected to produce 30 rows.
+TEST_F(NestedLoopJoinTest, parallel) {
+  auto leftVectors = makeBatches(10, 5, defaultLeftType_, pool_.get());
+  auto rightVectors = makeBatches(7, 5, defaultRightType_, pool_.get());
+  runTest(leftVectors, rightVectors, 5);
+}
 
-  auto left = {makeRowVector({sequence<int32_t>(2)})};
-  auto right = {makeRowVector({"u_c0"}, {sequence<int32_t>(3)})};
+TEST_F(NestedLoopJoinTest, bigintArray) {
+  auto leftVectors = makeBatches(1000, 5, defaultLeftType_, pool_.get());
+  auto rightVectors = makeBatches(900, 5, defaultRightType_, pool_.get());
+  createDuckDbTable("t", leftVectors);
+  createDuckDbTable("u", rightVectors);
 
   auto planNodeIdGenerator = std::make_shared<core::PlanNodeIdGenerator>();
-  CursorParameters params;
-  params.maxDrivers = 5;
-  params.planNode =
+  auto plan =
+      PlanBuilder(planNodeIdGenerator)
+          .values(leftVectors)
+          .nestedLoopJoin(
+              PlanBuilder(planNodeIdGenerator).values(rightVectors).planNode(),
+              "t0 = u0",
+              {"t0", "u0"},
+              core::JoinType::kFull)
+          .planNode();
+
+  assertQuery(plan, "SELECT t0, u0 FROM t FULL JOIN u ON t.t0 = u.u0");
+}
+
+TEST_F(NestedLoopJoinTest, allTypes) {
+  RowTypePtr leftTypes = ROW(
+      {{"t0", BIGINT()},
+       {"t1", VARCHAR()},
+       {"t2", REAL()},
+       {"t3", DOUBLE()},
+       {"t4", INTEGER()},
+       {"t5", SMALLINT()},
+       {"t6", TINYINT()}});
+
+  RowTypePtr rightTypes = ROW(
+      {{"u0", BIGINT()},
+       {"u1", VARCHAR()},
+       {"u2", REAL()},
+       {"u3", DOUBLE()},
+       {"u4", INTEGER()},
+       {"u5", SMALLINT()},
+       {"u6", TINYINT()}});
+
+  auto leftVectors = makeBatches(60, 5, leftTypes, pool_.get());
```

Comment:
Shall we test with all the join types supported? Thanks!

Reply:
Added, thanks!